commit 08d401e83000ed52f993415db916928b4ebebe37 (refs/changes/40/1740/2)
Author: Angela Fong <angela.fong@joyent.com>
Date:   2017-04-04T23:37:28+00:00 (2 years, 6 months ago)
    
    ADMINUI-2374 Support entry/display of network pool description
    Reviewed by: Marsell Kukuljevic

diff --git a/lib/network_pools.js b/lib/network_pools.js
index e53169c4..c0a76c99 100644
--- a/lib/network_pools.js
+++ b/lib/network_pools.js
@@ -14,6 +14,10 @@ module.exports.createNetworkPool = function (req, res, next) {
         networks: req.body.networks
     };
 
+    if (req.body.description && req.body.description.length) {
+        body.description = req.body.description;
+    }
+
     if (req.body.owner_uuids && req.body.owner_uuids.length) {
         body.owner_uuids = req.body.owner_uuids;
     }
@@ -48,6 +52,10 @@ module.exports.updateNetworkPool = function (req, res, next) {
     var uuid = req.params.uuid;
     var body = req.body || {};
 
+    if (body.description.length === 0) {
+        delete body.description;
+    }
+
     req.sdc[req.dc].napi.updateNetworkPool(uuid, body, function (err, job) {
         if (err) {
             req.log.fatal(err);
diff --git a/www/js/tpl/network-pools-form.hbs b/www/js/tpl/network-pools-form.hbs
index 74686045..0c223c03 100644
--- a/www/js/tpl/network-pools-form.hbs
+++ b/www/js/tpl/network-pools-form.hbs
@@ -31,6 +31,13 @@
                 </div>
             </div>
 
+            <div class="form-group">
+                <label class="control-label col-sm-4">Description</label>
+                <div class="controls col-sm-7">
+                    <input type="text" class="form-control" name="description" value="{{description}}" placeholder="network pool description (optional)">
+                </div>
+            </div>
+
             <div class="form-group">
                 <label class="control-label col-sm-4">Networks</label>
                 <div class="controls col-sm-7">
diff --git a/www/js/views/network-pools-form.js b/www/js/views/network-pools-form.js
index 84dd5ed5..ae51e3b7 100644
--- a/www/js/views/network-pools-form.js
+++ b/www/js/views/network-pools-form.js
@@ -175,6 +175,7 @@ module.exports = Backbone.Marionette.ItemView.extend({
         this.ui.saveButton.prop('disabled', true);
         this.stickit(this.networkPool, {
             'input[name=name]': 'name',
+            'input[name=description]': 'description',
             'input[name="owner_uuids[]"]': 'owner_uuids'
         });
         this.$('select').chosen({
